plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/CollationAwareQueryBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.postgresql;
+
+import io.trino.plugin.jdbc.DefaultQueryBuilder;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcJoinCondition;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+public class CollationAwareQueryBuilder
```

Comment:
I do not have a strong opinion, but it feels weird to have a `DefaultQueryBuilder` that handles collation for predicate pushdown and a `CollationAwareQueryBuilder` that handles collation of joins.

Reply:
Now that we have this abstraction maybe we can simplify the collation application for predicate pushdown as well. I consider the existing collation aware predicate pushdown to be a "hack" - we can maybe migrate it to the new mechanism as well (in separate PRs).

Reply:
Done.